gpu/gpu_info_cuda.c

```diff
@@ -27,6 +28,9 @@ void cuda_init(char *cuda_lib_path, cuda_init_resp_t *resp) {
       {"nvmlDeviceGetVbiosVersion", (void *)&resp->ch.nvmlDeviceGetVbiosVersion},
       {"nvmlDeviceGetBoardPartNumber", (void *)&resp->ch.nvmlDeviceGetBoardPartNumber},
       {"nvmlDeviceGetBrand", (void *)&resp->ch.nvmlDeviceGetBrand},
+      {"nvmlDeviceGetMigMode", (void *)&resp->ch.nvmlDeviceGetMigMode},
+      {"nvmlDeviceGetMigDeviceHandleByIndex", (void *)&resp->ch.nvmlDeviceGetMigDeviceHandleByIndex},
+      {"nvmlDeviceGetDeviceHandleFromMigDeviceHandle", (void *)&resp->ch.nvmlDeviceGetDeviceHandleFromMigDeviceHandle},
```

Comment:
I think we should add `nvmlDeviceGetMaxMigDeviceCount` and not hard-code the 8 device assumption.  Apparently the A100 supports 10 devices, and I know we have users on A100's.

Reply:
hmm, do you have a ref on that?  I believe the number is <= 7 per Nvidia docs.   I can also double check.

Reply:
https://www.nvidia.com/content/dam/en-zz/Solutions/design-visualization/solutions/resources/documents1/TB-10226-001_v01.pdf

On page 4:
```
NVIDIA vGPU software, which uses temporal partitioning, can partition a NVIDIA A100 up to 10
vGPUs, thereby 10 VM’s can access this shared resource (40 GB of GPU memory) with 4 GB
GPU memory allocated per VM
```

Reply:
Ok, so vGPU is in fact different than MIG, and also yes the upper limit to vGPU is 10 but the upper limit to MIG is 7 -- see page 5 your doc.